File: examples/animation-loop/gifexport.js
let duration = 5;
let margin = 3;

function setup(c) {
  c.size(40, 21);
  c.backgroundBg("197");
  c.saveCanvasToGIF("canvas.gif", duration);
}

function draw(c) {
  c.clear();
  c.stroke("RUNAL", "255", "197");
  let theta = c.loopAngle(duration);

  for (let y = margin; y < c.height - margin; y++) {
    let x = c.map(Math.sin(theta + y), -1, 1, margin, c.width - margin);

    c.point(x, y);
  }
}

function onKey(c, e) {
  if (e.key == "space") {
    c.noiseSeed(Date.now());
  }
}

File: examples/gif-export/gif-export.go
package main

import (
	"context"
	"math"
	"time"

	"github.com/emprcl/runal"
)

const (
	duration = 5
	margin   = 3
)

func main() {
	runal.Run(context.Background(), setup, draw, runal.WithOnKey(onKey))
}

func setup(c *runal.Canvas) {
	c.Size(40, 21)
	c.BackgroundBg("197")
	c.SaveCanvasToGIF("canvas.gif", duration)
}

func draw(c *runal.Canvas) {
	c.Clear()
	c.Stroke("RUNAL", "255", "197")
	theta := c.LoopAngle(duration)

	for y := margin; y < c.Height-margin; y++ {
		x := c.Map(math.Sin(theta+float64(y)), -1, 1, margin, float64(c.Width-margin))
		c.Point(int(x), y)
	}
}

func onKey(c *runal.Canvas, e runal.KeyEvent) {
	if e.Key == "space" {
		c.NoiseSeed(time.Now().Unix())
	}
}
